src/container/srv_target.c

```diff
@@ -795,18 +795,27 @@ cont_child_stop(struct ds_cont_child *cont_child)
 	}
 }
 
+static int ds_cont_child_close_all(uuid_t po_uuid);
+
 void
 ds_cont_child_stop_all(struct ds_pool_child *pool_child)
 {
 	d_list_t		*cont_list;
 	struct ds_cont_child	*cont_child;
+	int			 rc;
 
 	D_DEBUG(DF_DSMS, DF_UUID"[%d]: Stopping all containers\n",
 		DP_UUID(pool_child->spc_uuid),
 		dss_get_module_info()->dmi_tgt_id);
 
 	D_ASSERT(d_list_empty(&pool_child->spc_list));
 
+	/* Close potential leaked container open handlers by force. */
+	rc = ds_cont_child_close_all(pool_child->spc_uuid);
```

Comment:
It looks better to me to close all leaked container open handles when stopping a container (in cont_child_stop())

Reply:
Unless we link all related open handle to the cont_child, otherwise from the cont_child perspective, it does not know who are opening the container.